libs/ui-toolkit/datepicker/src/lib/datepicker.component.ts

```diff
@@ -0,0 +1,131 @@
+import { DatepickerContentComponent } from './datepicker-content.component';
+import { Component, AfterViewInit, Input, HostBinding, ViewEncapsulation, ViewChild, ChangeDetectionStrategy } from '@angular/core';
+import { ComponentType, ComponentPortal } from '@angular/cdk/portal';
+import { MatCalendar, MatCalendarView, MatDatepicker } from '@angular/material/datepicker';
+import { CalendarHeaderComponent } from './calendar-header.component';
+import { take } from 'rxjs/operators';
+
+export const SELECTOR = 'gor-datepicker';
+
+/**
+ * Component responsible for managing the datepicker popup/dialog.
+ *
+ * @link https://github.com/angular/components/blob/master/src/material/datepicker/datepicker.ts
+ */
+@Component({
+  selector: SELECTOR,
+  template: '',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  exportAs: 'gorDatepicker',
+  encapsulation: ViewEncapsulation.None
+})
+export class DatepickerComponent<D> extends MatDatepicker<D> {
+
+  private callPrivateMember<R>(p: string, ...args: any[]): R {
+    const m = this[p];
+
+    if (p in this) {
+      return this[p].call(this, ...args);
+    }
+
+    console.error(`Can't call inexistent ${p} private member MatDatepicker!`);
+  }
+
+  private getPrivateMember(p: string, allowFalsy = false) {
+    if (!(p in this) && !allowFalsy) {
+      console.error(`Member ${p} doesn't exist in MatDatepicker!`);
+    }
+
+    return this[p];
+  }
+
+  private setValueToPrivateMember(p: string, v: any) {
+    this[p] = v;
+  }
```

Comment:
callPrivateMember, getPrivateMember and setValueToPrivateMember seem highly generic. What do you think about turning them into reusable functions?

Reply:
Not yet necessary, I mean, turning easy to do dangerous things can be dangerous... If such thing is required again, we can do something about, the datepicker was a very specific case, sadly I didn't found better way without build all from ground up

Reply:
IMHO you did a good job (: